runtime/compiler/ilgen/Walker.cpp

```diff
@@ -2257,18 +2257,8 @@ TR_J9ByteCodeIlGenerator::calculateArrayElementAddress(TR::DataType dataType, bo
 
    if (comp()->getOption(TR_EnableSIMDLibrary))
        {
-       if (dataType == TR::VectorInt8)
-         dataType = TR::Int8;
-       else if (dataType == TR::VectorInt16)
-         dataType = TR::Int16;
-       else if (dataType == TR::VectorInt32)
-         dataType = TR::Int32;
-       else if (dataType == TR::VectorInt64)
-         dataType = TR::Int64;
-       if (dataType == TR::VectorFloat)
-         dataType = TR::Float;
-       else if (dataType == TR::VectorDouble)
-         dataType = TR::Double;
+       if (dataType.isVector())
+          dataType = dataType.getVectorElementType();
```

Comment:
can `getVectorElementType()` just return `this` if not a vector type?

Reply:
Interesting idea. I worry that there will be some hidden bugs due to that. I think currently, whenever we ask `getVectorElementType() `we indeed expect the type to be a vector. And actually, this method was not added now. It existed before. It used to return `TR::NoType` on non-vector types and now it asserts.

I would like to preserve the current (or more strict)behavior for now as much as possible. But we can definitely consider this change in the future.

Reply:
All right. It just seemed like a common pattern that could be simplified, but you should probably change the name of it if you're going to follow my suggestion anyway (since it's a little weird to ask a non-vector type for its vector element type, unless you consider all types to be vectors and scalars are just a degenerate vector. but i don't think that's how you consistently view them so...). Anyway, I can live with a follow-on change.